hedera-node/cli-clients/src/test/java/com.hedera.services.cli.sign.test/RecordStreamSigningUtilsTest.java

```diff
@@ -0,0 +1,181 @@
+    }
+
+    @Test
+    @DisplayName("Failed if record stream file is not version 6")
+    void cannotSignVersion5RecordStreamFile() {
+        final var signedFileDestination = Path.of(tmpDir.getPath() + "/2021-01-12T19_44_28.960705001Z.rcd_sig");
+        final Path fileToSign;
+        fileToSign = Path.of(Objects.requireNonNull(AccountBalanceSignCommandTest.class
+                        .getClassLoader()
+                        .getResource("com.hedera.services.cli.sign.test/2021-01-12T19_44_28.960705001Z.rcd"))
+                .getPath());
+
+        assertFalse(RecordStreamSigningUtils.signRecordStreamFile(
+                signedFileDestination, fileToSign, TestUtils.loadKey(), hapiVersion));
+    }
+
+    @Test
+    @DisplayName("Failed if hapi version is number format")
+    void failedToSignWithWrongHapiFormat() {
+        final var signedFileDestination = Path.of(tmpDir.getPath() + "/2022-09-19T21_09_17.348788413Z.rcd.gz_sig");
+        final Path fileToSign;
+        fileToSign = Path.of(Objects.requireNonNull(AccountBalanceSignCommandTest.class
+                        .getClassLoader()
+                        .getResource("com.hedera.services.cli.sign.test/2022-09-19T21_09_17.348788413Z.rcd.gz"))
+                .getPath());
+        final var hapiVersion = "a.b.c-test";
+
+        assertFalse(RecordStreamSigningUtils.signRecordStreamFile(
+                signedFileDestination, fileToSign, TestUtils.loadKey(), hapiVersion));
+    }
+
+    @Test
+    @DisplayName("Failed to generate signature file with invalid file")
+    void failedToSignWithInvalidFile() {
+        // given:
+        final var signedFileDestination = Path.of(tmpDir.getPath());
+        final Path fileToSign;
+        fileToSign = Path.of(Objects.requireNonNull(AccountBalanceSignCommandTest.class
+                        .getClassLoader()
+                        .getResource("com.hedera.services.cli.sign.test/2023-04-18T14_08_20.465612003Z.rcd"))
+                .getPath());
+
+        // then:
+        assertFalse(RecordStreamSigningUtils.signRecordStreamFile(
+                signedFileDestination, fileToSign, TestUtils.loadKey(), hapiVersion));
+    }
+
+    @Test
+    @DisplayName("Failed to generate signature file with empty record stream file")
+    void failedToSignWithEmptyRecordFile() {
+        final var signedFileDestination = Path.of(tmpDir.getPath() + "/2022-09-19T21_09_17.348788413Z.rcd.gz_sig");
+        final Path fileToSign = Path.of("testFile");
+        try (final var fos = new SerializableDataOutputStream(new FileOutputStream(fileToSign.toFile()))) {
+            fos.writeInt(RecordStreamSigningUtils.SUPPORTED_STREAM_FILE_VERSION);
+            fos.flush();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+
+        assertFalse(RecordStreamSigningUtils.signRecordStreamFile(
+                signedFileDestination, fileToSign, TestUtils.loadKey(), hapiVersion));
```

Comment:
This seems a incomplete file (with header) , not an empty file

Reply:
Yes, I did this on purpose. I wanted to pass the file version check and go to the next block of the code.

Reply:
This is for unit testing section of the code. Is it ok with you? If there are no other objections would you mind resolving the conversation?"